Stratis.Bitcoin/FullNode.cs

```diff
@@ -133,6 +143,13 @@ void RunLoop()
 						//TODO: 
 						Logs.FullNode.LogError("Block rejected: " + block.Error.Message);
 					}
+
+					if (!reorg && block.Error == null)
```

Comment:
making 1 request and saving the blocks are kind of cheap. I would have create a async method, where I query the BlockRepository to know if it contains the block, and if not, saving it. Not awaiting the method in the consensus loop for not slowing it.